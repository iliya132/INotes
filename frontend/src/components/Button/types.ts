import { MouseEventHandler } from "react";

export interface IButtonProps {
    title: string,
    className?: string,
    disabled?: boolean;
    onClick?: MouseEventHandler<HTMLButtonElement>;
}