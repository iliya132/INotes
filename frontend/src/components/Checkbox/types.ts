/* eslint-disable no-unused-vars */
export interface ICheckboxProps {
    defaultValue?: boolean;
    id: string;
    className?: string;
    onChanged? : (newValue:boolean) => void;
    title?: string;
}
