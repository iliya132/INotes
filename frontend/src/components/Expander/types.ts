export interface IExpanderProps {
    isExpanded: boolean;
    expanderBody: JSX.Element;
    children: JSX.Element;
    onClick?: () => void;
}
