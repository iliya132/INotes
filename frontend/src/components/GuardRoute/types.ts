export interface IGuardRouteProps {
    canActivate: boolean;
    redirectTo: string;
    children: JSX.Element;
    onSuccess?: () => void
  }
  