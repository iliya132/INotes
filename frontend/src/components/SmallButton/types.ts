import { Icons } from "../Svg/types";

export interface ISmallButtonProps {
    icon: Icons;
    className?: string;
    size?: number;
    disabled?: boolean;
    onClick?: () => void;
}
