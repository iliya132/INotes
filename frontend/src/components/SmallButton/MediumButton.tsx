import classNames from 'classnames';
import React from 'react';
import Svg from '../Svg';
import styles from './SmallButton.scss';
import { IMediumButtonProps } from './types';

export function MediumButton(props: IMediumButtonProps) {
    const { icon, size, className, onClick, title, disabled } = props;

    const handleOnClick = () => {
        if (disabled) {
            return;
        } else if (onClick) {
            onClick();
        }
    };

    return (
        <div className={classNames(styles['medium-container'], className)} onClick={handleOnClick}>
            <Svg icon={icon} className={styles['small-button-icon']} width={size} height={size} />
            <span>{title}</span>
        </div>
    );
}
