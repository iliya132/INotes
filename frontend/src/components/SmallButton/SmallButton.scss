@import '../../Misc/colors.scss';

.small-button-container {
    width: 40px;
    height: 40px;
    margin: 2px;
    border-radius: 5px;
    background-color: white;
    box-shadow: 0 4px 4px rgb(0 0 0 / 25%);
    text-align: center;
    vertical-align: middle;
    cursor: pointer;

    &:hover {
        background-color: rgb(233 233 233);
    }
}

.small-button-icon {
    display: block;
    height: 100%;
    margin: auto;
}

.medium-container{
    display: flex;
    flex-direction: row;
    height: 40px;
    box-sizing: border-box;
    margin: 2px;
    background-color: white;
    border-radius: 5px;
    box-shadow: 0 4px 4px rgb(0 0 0 / 25%);
    text-align: center;
    vertical-align: middle;
    cursor: pointer;
    padding: 5px 10px;

    &:hover{
        background-color: rgb(233 233 233);
    }

    & span{
        display: inline-block;
        margin-top: auto;
        margin-bottom: auto;
        margin-left: 5px;
    }
}
