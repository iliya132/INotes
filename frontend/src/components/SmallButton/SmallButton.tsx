import classNames from 'classnames';
import React from 'react';
import Svg from '../Svg';
import styles from './SmallButton.scss';
import { ISmallButtonProps } from './types';

export function SmallButton(props: ISmallButtonProps) {
    const { icon, size, className, onClick, tooltip, disabled } = props;

    const handleOnClick = () => {
        if (disabled) {
            return;
        } else if (onClick) {
            onClick();
        }
    };

    return (
        <div
            className={classNames(styles['small-button-container'], className)}
            onClick={handleOnClick}
            title={tooltip}
            data-testid={icon.toString()}>
            <Svg icon={icon} className={styles['small-button-icon']} width={size} height={size} />
        </div>
    );
}
