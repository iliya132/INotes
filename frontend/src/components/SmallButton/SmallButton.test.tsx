/* eslint-disable no-undef */
import React from 'react';
import { SmallButton } from './SmallButton';
import { act } from 'react-dom/test-utils';
import { Icons } from '../Svg/types';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/react';

describe('Test Button component', () => {
    it('can be rendered', () => {
        act(() => {
            render(<SmallButton icon={Icons.Dots} />);
        });

        let btn = document.body?.querySelector('.small-button-container');

        expect(btn!.innerHTML).toContain('#dots');
        expect(btn).toMatchSnapshot('button-snapshot');
    });

    it('can call events on click', async () => {
        let isPressed = false;
        const user = userEvent.setup();
        const testCallback = () => (isPressed = true);
        act(() => {
            render(<SmallButton icon={Icons.Dots} onClick={testCallback} />);
        });

        let btn = document.body?.querySelector('.small-button-container');
        expect(btn).not.toBeNull()

        await act(async () => {
            await user.click(btn!);
        })

        expect(isPressed).toEqual(true);
    });

    it('setup className', () => {
        const className = 'test-classname';
        const expectedClassName = 'small-button-container test-classname';
        act(() => {
            render(<SmallButton icon={Icons.Dots} className={className} />);
        });
        let btn = document.body.querySelector('.small-button-container');
        expect(btn).not.toBeNull();

        expect(btn!.className).toEqual(expectedClassName);
    });

    it('should be disabled when disabled props is set', async () => {
        const user = userEvent.setup();
        let isPressed = false;
        const testCallback = () => (isPressed = true);
        act(() => {
            render(<SmallButton icon={Icons.Dots} onClick={testCallback} disabled={true} />);
        });
        let btn = document.body?.querySelector('.small-button-container');
        expect(btn).not.toBeNull();

        await act(async () => {
            await user.click(btn!);
        });

        expect(isPressed).toEqual(false);
    });
});
