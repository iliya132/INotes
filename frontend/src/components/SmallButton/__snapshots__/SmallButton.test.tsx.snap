// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component can be rendered: button-snapshot 1`] = `
<div
  class="small-button-container"
  data-testid="dots"
>
  <svg
    class="image small-button-icon tooltip"
    fill="white"
    height="25"
    width="25px"
  >
    <use
      xlink:href="test-file-stub#dots"
    />
  </svg>
</div>
`;
